README.md

```diff
@@ -513,6 +513,32 @@ and [development](https://docs.ansible.com/ansible/latest/dev_guide/index.html).
   described in the 
   [Ansible docs](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html).
 
+## Collections
+
+### Modules and module_utils
+In a collection, the names of the module and module_utils files must be unique
+since all of them are located in
+`COLLECTION_NAMESPACE/COLLECTION_NAME/plugins/modules` (modules) and
+`COLLECTION_NAMESPACE/COLLECTION_NAME/plugins/module_utils` (module supporting
+files).  It is recommended to:
+* start module filenames with `ROLENAME_`
+* use a sub-directory `module_utils/ROLENAME/` for the role-specific
+  module_utils files
+* avoid generic names such as `size.py`
+
+If you are converting several roles into collection format and bundling them in
+a collection, you may need to rename files so that they are unique.  You may
+have to edit your module files to adjust the python imports.
+
+### Private Roles
+If your collection uses "private" roles - roles used to implement features in
+the public roles, but not meant to be used outside of the role - these roles are
+"top level" roles in the collection, meaning they can be used directly by users - there is no mechanism in Ansible to restrict role usage.  If you want to denote
+these roles as being "private" to the collection, and not supported for use
+directly, begin the name of the role with an underscore `_`.  This doesn't make
```

Comment:
This is unpossible. Ansible roles in a collection MUST adhere to the naming convention that they consist only of alphanumerics and the underscore character and they MUST begin with an alpha character. Galaxy will fail the import of the collection if this suggestion is implemented. The rules against this are hard coded into the Ansible Galaxy code checker and into ansible-lint to ensure that exactly this does not happen.

Reply:
Hmm - because it actually works to have roles in the collection with a name that begins with an underscore.  At least, we have been doing integration testing - we have not run ansible-lint against the converted collection, nor have we tried to import into Galaxy.  So it would seem the checkers are more restrictive than Ansible itself in this respect.
Do you have a link to the documentation where it says that the names of the roles in a collection must begin with alphabetic?  @nhosoi @pcahyna looks like we'll have to scrap that feature of the system roles conversion, and just document that the rsyslog and pcp roles should not be used directly.

Reply:
Here is the relevant bit from ansible-lint: https://github.com/ansible/ansible-lint/blob/master/lib/ansiblelint/rules/RoleNames.py#L29

Ansible-lint is part of the Galaxy import. So you're right - there's nothing in the Ansible engine that prevents underscores in the role names but you should find an error when you run Ansible Lint and Galaxy will fail on the import process.

Reply:
Sorry for having proposed something that is in conflict with Ansible docs. Still, would it be feasible to change the Ansible standards so that a leading underscore is allowed? We did this for variables in the past. Leading underscores for identifiers are common in programming languages.

Reply:
This is not the forum for that sort of change. That needs to be made by the core Ansible Galaxy team, if it is going to be made. It's not a standards restriction, at this point it's literally a technical imitation of the Ansible Galaxy importer.